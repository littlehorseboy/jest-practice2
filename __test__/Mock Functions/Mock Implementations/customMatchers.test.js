const myMockFn = jest
  .fn()
  .mockReturnValue('default')
  .mockImplementation(scalar => 42 + scalar)
  .mockName('add42');

test('call myMockFn', () => {
  expect(myMockFn(5)).toBe(5 + 42);
  expect(myMockFn(6)).toBe(6 + 42);
});

test('toBeCalled()', () => {
  expect(myMockFn).toBeCalled();
});

test('toBeCalledWith()', () => {
  expect(myMockFn).toBeCalledWith(5);
  expect(myMockFn).toBeCalledWith(6);
});

test('lastCalledWith()', () => {
  // expect(myMockFn).lastCalledWith(5);
  expect(myMockFn).lastCalledWith(6);
});

test('toMatchSnapshot()', () => {
  expect(myMockFn).toMatchSnapshot();
});

test('myMockFn was called at least once', () => {
  expect(myMockFn.mock.calls.length).toBeGreaterThan(0);
});

test('myMockFn was called at least once with the specified args', () => {
  expect(myMockFn.mock.calls[0]).toContain(5);
  expect(myMockFn.mock.calls[1]).toContain(6);
});

test('The last call to the mock function was called with the specified args', () => {
  expect(myMockFn.mock.calls[myMockFn.mock.calls.length - 1]).toEqual([6]);
});

test('The first arg of the last call to the mock function was `6`', () => {
  expect(myMockFn.mock.calls[myMockFn.mock.calls.length - 1][0]).toBe(6);
});

test('The mock function calls was toEqual `[[5], [6]]`', () => {
  expect(myMockFn.mock.calls).toEqual([[5], [6]]);
});

test('The mock function MockName is `add42`', () => {
  expect(myMockFn.getMockName()).toBe('add42');
});
