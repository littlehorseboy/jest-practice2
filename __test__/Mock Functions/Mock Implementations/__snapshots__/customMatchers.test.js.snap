// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchSnapshot() 1`] = `
[MockFunction add42] {
  "calls": Array [
    Array [
      5,
    ],
    Array [
      6,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": 47,
    },
    Object {
      "type": "return",
      "value": 48,
    },
  ],
}
`;
